### PR TITLE
us056: role bulk import

### PR DESCRIPTION
**Pull request by amordzon**

### CODE REVIEW

File: src/controllers/JobRoleController.ts [hunk 1]
```diff
@@ -83,7 +85,11 @@ export const getSingleJobRole = async (
           : jobRolesIds[0];
     }
 
-    const jobRole = await getJobRoleById(id, token);
+    const jobRole = await getJobRoleById(
+      id,
+      token,
+      'Failed to get job role details.',
```
Comment (ghost) [inline src/controllers/JobRoleController.ts:87]: That's probably not needed.

File: src/controllers/JobRoleController.ts [hunk 2]
```diff
@@ -98,3 +104,74 @@ export const getSingleJobRole = async (
...
+export const getJobApplyForm = async (
+  req: express.Request,
+  res: express.Response,
+): Promise<void> => {
+  try {
+    const currentId = parseInt(req.params.id, 10);
+    const jobRole = await getJobRoleById(
+      req.params.id,
+      req.session.token,
+      'Failed to get job apply form.',
```
Comment (ghost) [inline src/controllers/JobRoleController.ts]: likely not needed

File: src/services/JobRoleService.ts [hunk 3]
```diff
@@ -65,6 +66,62 @@ export const getJobRoleById = async (
...
+  }
+};
+
+export const postApplyFileForm = async (
+  token: string,
+  id: string,
+  file: Express.Multer.File,
+): Promise<JobRoleDetailsResponse> => {
+  try {
+    const maxSize = 5 * 1024 * 1024;
```
Comment (Alicja-Kainos) [inline src/services/JobRoleService.ts]: beautiful magic numbers :) please make it a nice method or constant ;)

File: src/services/JobRoleService.ts [hunk 4]
```diff
@@ -65,6 +66,62 @@ export const getJobRoleById = async (
...
+    throw new Error('Failed to post job apply form.');
+  }
+};
+
+export const postBulkImportJobRoles = async (
+  token: String,
+  file: Express.Multer.File,
+): Promise<void> => {
+  try {
+    const maxSize = 5 * 1024 * 1024;
```
Comment (Alicja-Kainos) [inline src/services/JobRoleService.ts]: also, see comment above

**Review by Alicja-Kainos — APPROVED** [review]

**Review by DariuszMech — APPROVED** [review]

**Pull request merged**